### ISSUE TITLE
QC guide for React 18 upgrade

### ISSUE THREAD

**Issue by bobular** [VEuPathDB/web-monorepo#1203]
This is a guide for how to test the sites following the [React 18 upgrade](https://github.com/VEuPathDB/web-monorepo/pull/1100) for the client code.

For checking if a bug was already on the live/production site:
* @aurreco-uga will provide a VectorBase dev site on the live/production site branch that will emulate live sites
* ortho, mbio and clinepi still have live production sites.

Outreach
- [ ] snackbars - do they still work and look OK? These can be tested as follows: 1. genomics site after changing organism preferences, 2. MapVEu when marker selections (click on markers) are made and various help is given, also notifications when selected markers go off-screen. (due to [these changes](https://github.com/VEuPathDB/web-monorepo/pull/1100/files?diff=split&w=1#diff-75d46e186753ef3fd833f668f13a804cf0cd898f789362e405422c95bd42f869R59-R80))
- [ ] do EDA viz thumbnails still all work (plotly plots, simple EDA/clinepi map viz, faceted plots, mbio networks etc) - because these are complex and brittle
- [ ] check the confirm dialogue for analysis deletion works https://qa.clinepidb.org/ce.b34/app/workspace/analyses (due to [these changes](https://github.com/VEuPathDB/web-monorepo/pull/1100/files?diff=split&w=1#diff-c1cfcd5b4619d5de436894119058759ea3118b93569f527d82c9785cedee4abeR62))
- [ ] check that legacy map redirects (e.g. 'view on map' links from the [release 64 notes](https://qa.vectorbase.org/vectorbase.b69/app/static-content/VectorBase/news.html#VectorBase64Released)) work. Not sure how we will test this on qa?? (code changes [here](https://github.com/VEuPathDB/web-monorepo/pull/1100/files?diff=split&w=1#diff-cf7aa46e34bae87cb9db6150a476222d4a4d6d55f46b0a86b3f92d1ab7eed82aR6-R62))
- [ ] check scroll-to-top button works - should be present in bottom-right (example below) on any website page after you scroll down a bit ([code change](https://github.com/VEuPathDB/web-monorepo/pull/1100/files?diff=split&w=1#diff-f769f771268f50ffad4c8a93576205132bec4f238ac951849700afbd1b9fa2c5R37)) ![image](https://github.com/user-attachments/assets/a7467c1d-e5e3-4964-be39-19d7872b3339)
- [ ] check behaviour of thumbnails at top of gene page (note that these are static images, not dynamic screenshots) ([code changes](https://github.com/VEuPathDB/web-monorepo/pull/1100/files?diff=split&w=1#diff-0c3f1df80e2e4a1c624cc9471c81b55e85ed223640bf38a707b98e8b7dc32a6eR73))

Developers
- [ ] we should enable https://react.dev/reference/react/StrictMode in all the dev servers and do testing while watching the console for errors and warnings (due to the use of the new React 18 Root API here https://github.com/VEuPathDB/web-monorepo/pull/1100/files?diff=split&w=1#diff-22503c1ed6527d2698b5d45ff08a9c39a5910f7fa21c278fe4408acae56dfa57R130 - and see discussion here https://github.com/reactwg/react-18/discussions/5) - note that we're not explicitly using new "concurrency" features but other behaviour may be subtly different
- [ ] explain and test `registerCustomElement` [here](https://github.com/VEuPathDB/web-monorepo/blob/486352728a86119b3fc73aea0b869ec1dbcb7409/packages/sites/genomics-site/webapp/wdkCustomization/js/client/components/common/Compound.jsx#L106) that is used on some record pages [here](https://github.com/VEuPathDB/web-monorepo/blob/486352728a86119b3fc73aea0b869ec1dbcb7409/packages/libs/web-common/src/component-wrappers/RecordPage.jsx#L16).  Presumably this is something to do with the image rendered near the top of this [compound record page](https://qa.vectorbase.org/vectorbase.b69/app/record/compound/CHEBI:18021).

**Comment by dmfalke** [comment]
> * explain and test `registerCustomElement` [here](https://github.com/VEuPathDB/web-monorepo/blob/486352728a86119b3fc73aea0b869ec1dbcb7409/packages/sites/genomics-site/webapp/wdkCustomization/js/client/components/common/Compound.jsx#L106) that is used on some record pages [here](https://github.com/VEuPathDB/web-monorepo/blob/486352728a86119b3fc73aea0b869ec1dbcb7409/packages/libs/web-common/src/component-wrappers/RecordPage.jsx#L16).  Presumably this is something to do with the image rendered near the top of this [compound record page](https://qa.vectorbase.org/vectorbase.b69/app/record/compound/CHEBI:18021).

Good idea. This was an early experiment to allow using custom html tags in the wdk model xml, to use react components.
